src/polytope_blmos.jl

```diff
@@ -37,6 +37,44 @@ function is_simple_linear_feasible(sblmo::CubeSimpleBLMO, v)
     return true
 end
 
+function is_decomposition_invariant_oracle_simple(sblmo::CubeSimpleBLMO)
+    lb = sblmo.lower_bounds
+    ub = sblmo.upper_bounds
+    return all(x -> (x[1] == 0 || x[1] == 1) && (x[2] == 0 || x[2] == 1), zip(lb, ub))
+end
+
+# After splitting, splitted variable will be fixed to either 0 or 1.
+function bounded_compute_inface_extreme_point(sblmo::CubeSimpleBLMO, direction, x, lb, ub, int_vars; kwargs...)
+    v = copy(x)
+    non_fixed_idx = findall(lb .!= ub)
+    non_fixed_int_idx = int_vars[non_fixed_idx]
+
+     # For non_fixed coordinates, zero-sum means that they are all fixed to origin.
+    sx = sum(x[non_fixed_int_idx])
+    if sx <= 0
+        return v
+    end
+    # Fix the point to the same face.
+    # Zero will be return only if d_i is greater than zero.
+    for idx in eachindex(direction)
+        if (idx in non_fixed_int_idx) && x[idx] > 0 
+            if x[idx] ≈ 1
+                v[idx] = 1
+            else
+                v[idx] = direction[idx] >  0 ? 0 : 1
+            end
+        end
+    end
+    return v       
+end
+
+# Once one variable is fixed to one, then there is only one feasible point.
+# Otherwise, assgin one to the entry corresponding to the smallest entry of d.
+# Same for the UnitSimplex and ProbabilitySimplex.
+function bounded_dicg_maximum_step(sblmo::CubeSimpleBLMO, x, direction, lb, ub, int_vars; kwargs...)
+    return FrankWolfe.dicg_maximum_step(FrankWolfe.ZeroOneHypercube(), x, direction)
```

Comment:
You probably cannot just use the Frank-Wolfe one since it does not know about the additional bounds.